test/testbufferoverrun.cpp

```diff
@@ -2168,6 +2170,15 @@ class TestBufferOverrun : public TestFixture {
         ASSERT_EQUALS("", errout.str());
     }
 
+    void array_index_enum_array() { // #8439
+        check("enum E : unsigned int { e1, e2 };\n"
+              "void f() {\n"
+              "    E arrE[] = { e1, e2 };\n"
+              "    arrE[sizeof(arrE)] = e1;\n"
+              "}");
+        ASSERT_EQUALS("[test.cpp:4]: (error) Array 'arrE[2]' accessed at index 8, which is out of bounds.\n", errout.str());
```

Comment:
Do we have a default configuration for testrunner (where `sizeof (unsigned int)==4`) ?